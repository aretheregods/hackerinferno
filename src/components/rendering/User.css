#user-container  {
    max-width: 100%;
    margin: 1vh 2vw 2vh 2vw;
    text-align: center;
}

#user-about {
    max-width: 90%;
}

#hacker-links a {
    text-decoration: none;
    color: red;
}

#hacker-links a:first-child {
    margin-left: 0px;
}
