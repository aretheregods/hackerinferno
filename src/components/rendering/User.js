import { DataContainer } from '../data/Data-Container';
import { hn_api_pages } from '../constants/API';

function HackerUser(props) {

    const data = props.component_data;
    const hacker_submitted = 'https://news.ycombinator.com/submitted?id=';
    const hacker_commented = 'https://news.ycombinator.com/threads?id=';
    const hacker_favourites = 'https://news.ycombinator.com/favorites?id='


    const user_data = (
        <div id="user-container">
            <h3 id="user-id">{data.id}</h3>
            <div id="user-info">
                <div id="about-container" dangerouslySetInnerHTML={{__html: data.about}}></div>
                <div id="user-data-container">
                    <p>joined: {data.created} | karma: {data.karma}</p>
                </div>
                <div id="hacker-links">
                    <a id="hacker-submitted" href={`${hacker_submitted}${data.id}`}>submitted | </a>
                    <a id="hacker-commented" href={`${hacker_commented}${data.id}`}>commented | </a>
                    <a id="hacker-favourites" href={`${hacker_favourites}${data.id}`}>favourites</a>
                </div>
            </div>
        </div>
    )

    return user_data;

}

export default function User({params}) {

    return (
        <DataContainer urlobj={hn_api_pages} endpoint="user" page={params.page}>
            {({component_data}) => <HackerUser component_data={component_data}/>}
        </DataContainer>
    )
}
