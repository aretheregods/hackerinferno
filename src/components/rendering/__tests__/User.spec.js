import { render } from 'inferno';
import { renderToSnapshot } from 'inferno-test-utils';
import { HackerUser } from '../User';

const data = {
    id: "something_whatever",
    about: "<p>It doesn't matter.</p>",
    created: "2 years ago",
    karma: 98302
};

it('should render "div" tag with information supplied by an object with user data', () => {
    const snapshot = renderToSnapshot(
        <HackerUser component_data={data} />
    )
    expect(snapshot).toMatchSnapshot();
})
