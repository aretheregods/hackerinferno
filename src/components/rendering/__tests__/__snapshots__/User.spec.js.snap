// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render "div" tag with information supplied by an object with user data 1`] = `
<div
  id="user-container"
>
  <h3
    id="user-id"
  >
    something_whatever
  </h3>
  <div
    id="user-info"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>It doesn't matter.</p>",
        }
      }
      id="about-container"
    />
    <div
      id="user-data-container"
    >
      <p>
        joined: 
        2 years ago
         | karma: 
      </p>
    </div>
    <div
      id="hacker-links"
    >
      <a
        href="https://news.ycombinator.com/submitted?id=something_whatever"
        id="hacker-submitted"
      >
        submitted | 
      </a>
      <a
        href="https://news.ycombinator.com/threads?id=something_whatever"
        id="hacker-commented"
      >
        commented | 
      </a>
      <a
        href="https://news.ycombinator.com/favorites?id=something_whatever"
        id="hacker-favourites"
      >
        favourites
      </a>
    </div>
  </div>
</div>
`;
